lib/matplotlib/axes/_axes.py

```diff
@@ -4559,6 +4559,10 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             collection.set_cmap(cmap)
             collection.set_norm(norm)
             collection._scale_norm(norm, vmin, vmax)
+        elif any({cmap, norm, vmin, vmax}):
```

Comment:
We probably should check `is not None` here rather than the truthyness of value because `vmin=0` will not trip this!